### ISSUE TITLE
Make/run integrated tests in parallel

### ISSUE THREAD

**Issue by johnomotani** [boutproject/BOUT-dev#1451]
It would be nice if `tests/integrated/test_suite_make` and `tests/integrated/test_suite` could run jobs in parallel: i.e. parallelize the loop at https://github.com/boutproject/BOUT-dev/blob/f334b72a44e8b0eced5fb577477dc01ea4c2b8ee/tests/integrated/test_suite#L105
... maybe have a command line argument to pass the number of processors to use, and get the makefile to pass a suitable number when e.g. `make -j # check-integrated-tests` is run.

**Comment by ZedThree** [comment]
What does the output look like?

**Comment by dschwoerer** [comment]
as if you would run `./test_suite -l test` for each test in serial.

Doing it in python would be nicer ...

**Comment by ZedThree** [comment]
`ctest` can also do this out of the box with `-j N`.

**Comment by dschwoerer** [comment]
Does ctest know how many cores each tests needs and takes that into account?

**Comment by dschwoerer** [comment]
The main reason I haven't implemented this, is because some tests run in parallel. If we additionally schedule more tests, this can make it worse. The above timings are from a machine with 4 physical cores, so you would normally run make with 8 threads. If we just take that number and pass it to parallel, this does not help, and it seems 2 is the optimal number.

I wanted to code something in python that reads from each file how many cores it needs, so it can be smart about scheduling tests. But I haven't gotten around to do it -.-

**Comment by ZedThree** [comment]
I agree, probably more trouble than it's worth. Won't fix?

**Comment by dschwoerer** [comment]
No strong preference, I might come around in the future.
Especially if I would get a system with more cores, the benefit would increase. However closing this issue will not prevent me from doing it :-)

**Comment by ZedThree** [comment]
Ok, closing for now. Currently sort of possible, just maybe not advisable.
